File: infrastructure/scripts/check-pltr-marketcap.js
const https = require('https');

async function getYahooFinanceData(ticker) {
  return new Promise((resolve, reject) => {
    const options = {
      hostname: 'query1.finance.yahoo.com',
      path: `/v8/finance/chart/${ticker}`,
      method: 'GET',
      headers: {
        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36',
        'Accept': 'application/json',
        'Accept-Language': 'en-US,en;q=0.9',
      }
    };

    const req = https.request(options, (res) => {
      let data = '';
      res.on('data', (chunk) => {
        data += chunk;
      });
      res.on('end', () => {
        try {
          const result = JSON.parse(data);
          if (result.chart && result.chart.result && result.chart.result[0]) {
            const meta = result.chart.result[0].meta;
            resolve({
              marketCap: meta.marketCap,
              price: meta.regularMarketPrice,
              currency: meta.currency
            });
          } else {
            reject(new Error('Invalid response structure'));
          }
        } catch (error) {
          console.error('Raw response:', data);
          reject(error);
        }
      });
    });

    req.on('error', (error) => {
      reject(error);
    });

    req.end();
  });
}

async function main() {
  console.log('Fetching PLTR market cap from Yahoo Finance...\n');
  
  try {
    const data = await getYahooFinanceData('PLTR');
    
    console.log('✅ PLTR Data:');
    console.log(`   Stock Price: $${data.price.toFixed(2)}`);
    console.log(`   Market Cap: $${(data.marketCap / 1e9).toFixed(2)}B`);
    console.log(`   Market Cap (raw): ${data.marketCap}`);
    console.log(`   Currency: ${data.currency}`);
    
  } catch (error) {
    console.error('❌ Error:', error.message);
  }
}

main();

File: infrastructure/scripts/check-pltr-marketcap.py
#!/usr/bin/env python3
import json
try:
    import yfinance as yf
except ImportError:
    print("Installing yfinance...")
    import subprocess
    subprocess.check_call(['pip', 'install', 'yfinance', '-q'])
    import yfinance as yf

# Fetch PLTR data
ticker = yf.Ticker("PLTR")
info = ticker.info

print(f"\n✅ PLTR Market Data:")
print(f"   Stock Price: ${info.get('currentPrice', info.get('regularMarketPrice', 'N/A'))}")
print(f"   Market Cap: ${info.get('marketCap', 0) / 1e9:.2f}B")
print(f"   Market Cap (raw): {info.get('marketCap', 0)}")
print(f"   Shares Outstanding: {info.get('sharesOutstanding', 0):,}")
